src/adv.py

```diff
@@ -49,3 +54,16 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+player_input = None
+while (player_input is not 'q'):
+    print(f"\nCurrent room: {player.room.name}\n")
+    print(f"{player.room.description}\n")
+    print(f"Where would you like to move {player.name}?\n")
+    print('[n] = North [s] = South [e] = East [w] = West [q] = Quit')
+    player_input = input()
+    previous_room = player.room
+    player.move(player_input)
+    if player.room == None:
+        print('\nCannot go that way\n')
+        player.room = previous_room
```

Comment:
AH HAHAHAHA! Not bad at all.

You could also check the player_input and append that to the player.room, see if that resolves to `None`, and check it that way too, but I like this!